### PR TITLE
Add roundtrip tests for new serializers [R4]

### LINKED ISSUE #2154: Create roundtrip unit tests for new Serializers
Create a new set of unit tests that will test the new Serializers.

So we take the examples.zip as input for roundtripping: xml -> json -> xml and
examples-json for json -> xml -> json.

### PR DESCRIPTION
**Pull request by mmsmits**
fixes: [#2154](https://github.com/FirelyTeam/firely-net-sdk/issues/2154)

### CODE REVIEW

File: src/Hl7.Fhir.Serialization.Tests/RoundtripNewSerializers.cs [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+        private static readonly string inputFolder = "input";
+        private static readonly string intermediate1Folder = "intermediate1";
+        private static readonly string intermediate2Folder = "intermediate2";
+
+        [DynamicData(nameof(prepareExampleZipFilesXml), DynamicDataSourceType.Method, DynamicDataDisplayName = nameof(GetTestDisplayNames))]
+        [DataTestMethod]
+        [TestCategory("LongRunner")]
+        public void FullRoundtripOfAllExamplesXmlNewSerializer(string file, FhirXmlPocoSerializer xmlSerializer, FhirXmlPocoDeserializer xmlDeserializer, JsonSerializerOptions jsonOptions)
+        {
+            var baseTestPath = Path.Combine(Path.GetTempPath(), xmlTestFolder);
```
Comment (marcovisserFurore) [inline src/Hl7.Fhir.Serialization.Tests/RoundtripNewSerializers.cs:31]: Can this be done in the method `prepareExampleZipFiles`? Then it is done only once, instead for every file.

File: src/Hl7.Fhir.Serialization.Tests/RoundtripNewSerializers.cs [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+            doRoundTrip(baseTestPath, file, xmlSerializer, xmlDeserializer, jsonOptions);
+        }
+
+        [DynamicData(nameof(prepareExampleZipFilesJson), DynamicDataSourceType.Method, DynamicDataDisplayName = nameof(GetTestDisplayNames))]
+        [DataTestMethod]
+        [TestCategory("LongRunner")]
+        public void FullRoundtripOfAllExamplesJsonNewSerializer(string file, FhirXmlPocoSerializer xmlSerializer, FhirXmlPocoDeserializer xmlDeserializer, JsonSerializerOptions jsonOptions)
+        {
+            var baseTestPath = Path.Combine(Path.GetTempPath(), jsonTestFolder);
```
Comment (marcovisserFurore) [inline src/Hl7.Fhir.Serialization.Tests/RoundtripNewSerializers.cs:40]: Can this be done in the method `prepareExampleZipFiles`? Then it is done only once, instead for every file.

**Review by marcovisserFurore — CHANGES_REQUESTED** [review]
Well done! Looks nice and much more clearer.

Some small things to think about.

**Comment by mmsmits** [comment]
Replaced with this PR: https://github.com/FirelyTeam/firely-net-sdk/pull/2347

**Pull request closed without merging**